### ISSUE TITLE
I find a problem which  can be repetitioned easily in the latest version, I'm sure there's something wrong with the framework

### ISSUE THREAD

**Issue by xianglei3** [open-mmlab/mmdetection#1904]
I had used the latest version for five weeks, and  the framework had worked well with standard data coco sets and default parameters. however, it hadn't done well in the following scenarios:

the data is : [https://tianchi.aliyun.com/competition/entrance/231763/information](url)
also  you can get the data from:
[http://tianchi-competition.oss-cn-hangzhou.aliyuncs.com/231763/round1/chongqing1_round1_train1_20191223.zip](url)

I had used "python  tools/train.py  ./configs/ssd300_coco.py"  
I chooosed default learningrate:2e-3:"optimizer = dict(type='SGD', lr=2e-3"
besides  i choose :     imgs_per_gpu=1, workers_per_gpu=1
When I use the default learning rate 2e-3 and above parameters, the following errors occur : 
 workflow: [('train', 1)], max: 24 epochs
/pytorch/aten/src/THC/THCTensorScatterGather.cu:131: void THCudaTensor_scatterKernel(TensorInfo<Real, IndexType>, TensorInfo<Real, IndexType>, TensorInfo<long, IndexType>, int, IndexType) [with IndexType = unsigned int, Real = float, Dims = 1]: block: [0,0,0], thread: [0,0,0] Assertion `indexValue >= 0 && indexValue < tensor.sizes[dim]` failed.

 At first, I suspected that it was related to data tagging errors, but it was not the data problem after multiple communication and verification .

 A very important phenomenon is that when I change my learning rate to a lower level 2e-4,  the error disappears, so on the surface, it has something to do with the learning rate 

the question is that changing the learning rate to a lower level just avoids the problem but doesn't really solve the problem, so I want to know the real reason, otherwise I will be very confused when using mmdecision in the future 

you can contact me by 83656313@qq.com

**Comment by hellock** [comment]
Different datasets may not fit the same hyper-parameters.  With improper settings, the model will diverge and errors occur.
@yhcao6 You may add more assertions and error info for SSD.

**Comment by yhcao6** [comment]
From your description, you use the default learning rate 2e-3 but modify the `imgs_per_gpu` from 8 to 1. The above error is expected and I guess there is NaN in the network prediction.
In such case you need modify the learning rate. Please refer to the [documentation](https://github.com/open-mmlab/mmdetection/blob/master/docs/GETTING_STARTED.md#train-a-model) and it is marked with Important. 

Besides, different datasets may need different hyper-parameters. You may try different setting to find a good one.
